src/ms-graph/client.ts

```diff
@@ -125,6 +126,33 @@ export class GraphClient {
     } while (nextLink);
   }
 
+  /**
+   * Wraps callApi with retry logic.
+   * @param link
+   * @param query
+   * @protected
+   */
+  protected async callApiWithRetry<T>({
+    link,
+    query,
+  }: {
+    link: string;
+    query?: QueryParams;
+  }): Promise<GraphClientResponse<T> | undefined> {
+    return retry(() => this.callApi<T>({ link, query }), {
+      maxAttempts: 3,
+      delay: 30_000,
+      timeout: 180_000,
+      factor: 2,
+      handleError: (error, attemptContext) => {
+        this.logger.info(
+          { error, attemptContext },
```

Comment:
Maybe would be helpful to log with the `link` if it doesn't include any sensitive data?